src/precice/bindings/python/precice/precice.py

```diff
@@ -57,8 +57,9 @@ def get_data_id(self, data_name, mesh_id):
         return self.interface.get_data_id(data_name, mesh_id)
 
     def set_mesh_vertices(self, mesh_id, size, positions, ids):
-        assert(positions.size == size * self.get_dimensions())
-        out_ids = self.interface.set_mesh_vertices(mesh_id, positions)
+        dimensions = self.get_dimensions()
+        assert(positions.size == size * dimensions)
+        out_ids = self.interface.set_mesh_vertices(mesh_id, positions.reshape((size, dimensions)))
```

Comment:
Maybe this is a stupid comment, but don't you have to always do the inverse operation? So `flatten` here instead of `reshape`?

Reply:
For the `vertex_ids` (scalar valued) we only get a array of dimension `n`. `flatten` and `reshape` are only needed for vector valued quantities like `positions` or `vector_data`.